tests/test_sft_trainer.py

```diff
@@ -0,0 +1,291 @@
+
+        model_args, data_args, training_args, tune_config = causal_lm_train_kwargs(
+            lora_target_modules
+        )
+        sft_trainer.train(model_args, data_args, training_args, tune_config)
+
+        # validate lora tuning configs
+        _validate_training(tempdir)
+        checkpoint_path = os.path.join(tempdir, "checkpoint-5")
+        adapter_config = _get_adapter_config(checkpoint_path)
+        _validate_adapter_config(adapter_config, "LORA", BASE_LORA_KWARGS)
+        for module in lora_target_modules["target_modules"]:
+            assert module in adapter_config.get("target_modules")
+
+
+def test_run_train_lora_target_modules_all_linear():
+    """Check runs lora tuning with all linear target modules."""
+    with tempfile.TemporaryDirectory() as tempdir:
+        lora_target_modules = copy.deepcopy(BASE_LORA_KWARGS)
+        lora_target_modules["output_dir"] = tempdir
+        lora_target_modules["target_modules"] = ["all-linear"]
+
+        model_args, data_args, training_args, tune_config = causal_lm_train_kwargs(
+            lora_target_modules
+        )
+        sft_trainer.train(model_args, data_args, training_args, tune_config)
+
+        # validate lora tuning configs
+        _validate_training(tempdir)
+        checkpoint_path = os.path.join(tempdir, "checkpoint-5")
+        adapter_config = _get_adapter_config(checkpoint_path)
+        _validate_adapter_config(adapter_config, "LORA", BASE_LORA_KWARGS)
+        llama_expected_modules = [
+            "o_proj",
+            "q_proj",
+            "gate_proj",
+            "down_proj",
+            "k_proj",
+            "up_proj",
+            "v_proj",
+        ]
+        for module in llama_expected_modules:
+            assert module in adapter_config.get("target_modules")
+
+
+def test_run_causallm_ft_and_inference():
+    """Check if we can bootstrap and finetune tune causallm models"""
+    with tempfile.TemporaryDirectory() as tempdir:
+        BASE_FT_KWARGS["output_dir"] = tempdir
+        model_args, data_args, training_args, tune_config = causal_lm_train_kwargs(
+            BASE_FT_KWARGS
+        )
+        sft_trainer.train(model_args, data_args, training_args, tune_config)
+
+        # validate ft tuning configs
+        _validate_training(tempdir)
+        checkpoint_path = os.path.join(tempdir, "checkpoint-5")
+        adapter_config = _get_adapter_config(checkpoint_path)
+        _validate_adapter_config(adapter_config, "PROMPT_TUNING", BASE_FT_KWARGS)
+
+        # Load the model
+        loaded_model = TunedCausalLM.load(checkpoint_path)
+
+        # Run inference on the text
+        output_inference = loaded_model.run(
+            "### Text: @NortonSupport Thanks much.\n\n### Label:", max_new_tokens=50
+        )
+        assert len(output_inference) > 0
+        assert "### Text: @NortonSupport Thanks much.\n\n### Label:" in output_inference
+
+
+def _validate_training(tempdir):
```

Comment:
hmm why not pass a flag here for validation dataset as bool. if set to True, also verify validation_loss.jsonl exists